### ISSUE TITLE
Don't recompress images when drag & drop from browser window

### ISSUE THREAD

**Issue by migozz** [telegramdesktop/tdesktop#25057]
### Steps to reproduce

1. Drag & drop image from browser window to telegram
2. The image is being recompressed, regardless of compress image settings

### Expected behaviour

Drag & drop image from browser window to telegram:

gif animated (uncompressed) -> gif original, not converted
gif animated (compressed) -> probably converted to mp4, not resized
gif animated (paste URL into textbox) -> not resized, converted to mp4?

png uncompressed -> png original, not recompressed
png compressed -> png compressed, preserves transparency?
png (paste URL into textbox)  -> all pics to be saved by right click -> save as...?

jpg uncompressed -> jpg original, not recompressed
jpg compressed -> jpg compressed
jpg (paste URL into textbox)  -> all pics to be saved by right click -> save as...?

### Actual behaviour

Drag & drop image from browser window to telegram:

gif animated (uncompressed) -> png, single frame
gif animated (compressed) -> jpg, single frame
gif animated (paste URL into textbox) -> resized to 320px width, converted to mp4, can be saved by right click -> save as...

png uncompressed -> png recompressed, loses transparency
png compressed -> jpg, resized to 1280px width, loses transparency
png (paste URL into textbox)  -> some pics recompressed, can be saved by right click -> save as..., on click opens in telegram
- some other pics can't be saved, on click opens in browser
example of file that can't be saved: https://upload.wikimedia.org/wikipedia/commons/4/47/PNG_transparency_demonstration_1.png

jpg uncompressed -> png recompressed, loses metadata
jpg compressed -> jpg, resized to 1280px width
jpg (paste URL into textbox)  -> some pics recompressed, can be saved by right click -> save as..., on click opens in telegram, 
- some other pics can't be saved, on click opens in browser
example of file that can't be saved: https://upload.wikimedia.org/wikipedia/commons/b/b2/JPEG_compression_Example.jpg

### Operating system

Windows 7

### Version of Telegram Desktop

4.1.1

### Installation source

Static binary from official website

**Comment by ilya-fedin** [comment]
Telegram has the following preference when pasting images from clipboard:
1. Local files
2. Image data
OSes usually have  a limited set of supported formats for image data and I don't think animated gifs is one of them. Apparently, your browser doesn't provide a path to local copy of the file and tdesktop (can't speak for Qt or other libraries tdesktop uses though) doesn't (and won't) support downloading content from remote URLs provided by clipboard.

TL;DR this behavior is most likely the intended one unless there's a bug in your browser.

**Comment by migozz** [comment]
Why it "won't" support downloading the content when "pasting" images (not pasting, but dragging)?
Why, at the same time, it does support downloading the content (without a permission) when paste URL into a textbox?

The file is taken from browser's cache on disk.
If I drag and drop the image to Windows Explorer, it comes up as the correct file (not downloaded from a remote URL).
This is a Telegram issue that it can't do that.

Not "closed" and not "completed".
The other questions are not answered (e.g. why it downscales the GIF, etc.).

**Comment by ilya-fedin** [comment]
> Why it "won't" support downloading the content when "pasting" images (not pasting, but dragging)?

Well, I don't know why, the only thing I know is there's no plan to support such a thing. Dragging is pasting, too.

> Why, at the same time, it does support downloading the content (without a permission) when paste URL into a textbox?

No, there's no such code, at least in tdesktop. If tdesktop does that, that's apparently a bug or a feature of Qt.

> The file is taken from browser's cache on disk.

The issue sounds like browser doesn't provide such URL to clipboard, can you check with some kind of clipboard inspector (if there's any for Windows)? The fact gifs become single frame, png is recompressed, jpeg becomes png: that all sounds like the browser supplies only image data (when the copying program converts the image to a lot of formats and it's not known which format is the original one, so Qt chooses the best one it thinks of: png).

> If I drag and drop the image to Windows Explorer, it comes up as the correct file (not downloaded from a remote URL).

Maybe browser uses some advanced URL format that is supported by explorer, but not supported by Qt and therefore not detected as a local URL? Unless you would find a way to extract what the browser sets in the clipboard, I don't think there will be any progress.

> e.g. why it downscales the GIF, etc.

Generation of link preview is server side. If you have any problems with it, it's a server side bug.

> This is a Telegram issue that it can't do that.

Well, it's how clipboard works, this may be not really convenient, but this is not a bug.
